python.d/python_modules/bases/FrameworkServices/SocketService.py

```diff
@@ -249,6 +273,28 @@ def _parse_config(self):
             except (KeyError, TypeError):
                 self.debug('No port specified. Using: "{0}"'.format(self.port))
 
+            self.ssl = bool(self.configuration.get('ssl', self.ssl))
+            if self.ssl and not _SSL_SUPPORT:
+                self.warning('SSL requested but not SSL module found, disabling SSL support.')
+                self.ssl = False
+            if _SSL_SUPPORT and not self.ssl:
+                self.debug('No SSL preference specified, not using SSL.')
+
+            if self.ssl and _SSL_SUPPORT:
+                self.key = str(self.configuration.get('ssl_key'))
+                self.cert = str(self.configuration.get('ssl_cert'))
```

Comment:
i believe `str` is not needed. How else can you set the key/cert? 
And `str(None)` is always True.